src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -652,11 +652,66 @@ public void CleanupAfterEachTest()
                 message: "DAB did not properly set the 'Application Name' connection string property.");
         }
 
+        /// <summary>
+        /// Validates that DAB supplements the PgSQL database connection strings with the property "ApplicationName" and
+        /// 1. Adds the property/value "Application Name=dab_oss_Major.Minor.Patch" when the env var DAB_APP_NAME_ENV is not set.
+        /// 2. Adds the property/value "Application Name=dab_hosted_Major.Minor.Patch" when the env var DAB_APP_NAME_ENV is set to "dab_hosted".
+        /// (DAB_APP_NAME_ENV is set in hosted scenario or when user sets the value.)
+        /// NOTE: "#pragma warning disable format" is used here to avoid removing intentional, readability promoting spacing in DataRow display names.
+        /// </summary>
+        /// <param name="configProvidedConnString">connection string provided in the config.</param>
+        /// <param name="expectedDabModifiedConnString">Updated connection string with Application Name.</param>
+        /// <param name="dabEnvOverride">Whether DAB_APP_NAME_ENV is set in environment. (Always present in hosted scenario or if user supplies value.)</param>
+        #pragma warning disable format
+        [DataTestMethod]
+        [DataRow("Host=foo;Username=testuser;", "Host=foo;Username=testuser;Application Name=", false, DisplayName = "[PGSQL|DAB OSS]:Addition of 'ApplicationName' property to connection string.]")]
+        [DataRow("Host=foo;Username=testuser;", "Host=foo;Username=testuser;Application Name=", true, DisplayName = "[PGSQL|DAB OSS]:Addition of 'ApplicationName' property to connection string.]")]
+        [DataRow("Host=foo;Username=testuser;Application Name=UserAppName", "Host=foo;Username=testuser;Application Name=UserAppName,", false, DisplayName = "[PGSQL|DAB OSS]:Addition of 'ApplicationName' property to connection string.]")]
+        [DataRow("Host=foo;Username=testuser;Application Name=UserAppName", "Host=foo;Username=testuser;Application Name=UserAppName,", true, DisplayName = "[PGSQL|DAB OSS]:Addition of 'ApplicationName' property to connection string.]")]
```

Comment:
DisplayNames should be unique and describe what the test does. Similar to L610-L615